openedx_events/data.py

```diff
@@ -47,36 +47,36 @@ class EventsMetadata:
             lexical ordering of strings (e.g. '0.9.0' vs. '0.10.0').
     """
 
-    id = attr.ib(type=UUID, init=False)
+    id = attr.ib(
+        type=UUID, default=None,
+        converter=attr.converters.default_if_none(attr.Factory(lambda: uuid1())),  # pylint: disable=unnecessary-lambda
+        validator=attr.validators.instance_of(UUID),
```

Comment:
**Note to Reviewer**: I also added type validators to ensure the data being sent is appropriate.
Please mark resolved once read by main reviewer.

Reply:
[question] does type=UUID not do this for us? same for all the other updated fields below

Reply:
Correct. It doesn't do that for us. Not sure if there is another way, but `id="mock-id"` works without the validator.

Reply:
Sigh